src/Rule/Uuid.php

```diff
@@ -23,17 +21,41 @@ class Uuid extends Regex
     const INVALID_UUID = 'Uuid::INVALID_UUID';
 
     /**
-     * UUID Version 4.
+     * UUID NIL & version binary masks
      */
-    const UUID_V4 = 4;
+    const UUID_NIL = 0b000001;
+    const UUID_V1  = 0b000010;
+    const UUID_V2  = 0b000100;
+    const UUID_V3  = 0b001000;
+    const UUID_V4  = 0b010000;
```

Comment:
I wrote the above in a bit of a hurry last friday. Let me add to what I meant:

- With the first sentences I meant that in my PR each bit's location maps to the version. The rightmost bit at the location zero maps to NIL (1 being 2^1), position 1 maps to V1 (2 being 2^2), position 3 to V3 (4 being 2^3). But while that looks better, and is easier to understand. It's not really necessary to have the mapping like that. Switching V3 and V4 would keep BC and only look illogical to someone studying the value of the constants.
- With the second paragraph the first sentence is a bit off. What I meant to say is that currently only V4 is supported, so it has very little meaning currently.

Personally when looking at the code I prefer the mapping to match the bit-order. But from a BC perspective I can imagine choosing for the option to switch V3 & V4 around and explaing that in comments within the code.